crates/language/src/target_language.rs

```diff
@@ -173,6 +185,88 @@ impl PatternLanguage {
         Self::get_language_with_parser(&mut parser, src)
     }
 
+    fn get_aliases(&self) -> &'static [&'static str] {
+        for (variant, aliases) in LANGUAGE_ALIASES {
+            if variant == self {
+                return aliases;
+            }
+        }
+        &[]
+    }
+
+    fn map_value(s: String) -> Self {
```

Comment:
I think we could safely just delegate this to from_string and always support aliases inside from_string.

Reply:
ok can do. I thought it was a bit different because the signature is this and we don't have a flavor when we are parsing CLI args:
```rust
pub fn from_string(name: &str, flavor: Option<&str>) -> Option<Self> {
```

so there may be a slightly different behaviour in terms of parsing preference/order but this probably links to / aligns with your other comments anyway (preserve the skipped variants, get rid of `to_str` and use the `impl fmt::Display for PatternLanguage` etc.)

but I can certainly combine them like I did inside of this `map_value` function and just essentially use None for aliases / extensions:
```rust
        if let Some(lang) = Self::from_string(&s, None) {
            return lang;
        }
```

Reply:
Yeah, `None` is fine (since aliases should just use the default flavor).

Reply:
done in: 0f1b3ce